In [16]:
import pandas as pd

# =========================
# Task 1: Load the Dataset
# =========================

# Direct download link
url = "https://drive.google.com/uc?export=download&id=1Mlu9r3afSGz5_02Y3wDJoQLpXszQXaIU"

# Load dataset with order_id as index
df = pd.read_csv(url, index_col='order_id')

print("First 5 rows:")
display(df.head())

print("\nLast 5 rows:")
display(df.tail())


First 5 rows:


,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1001,Alice Johnson,Laptop,1.0,85000.0,2024-01-15,delivered
1002,Bob Smith,Mouse,3.0,499.0,2024-01-16,pending
1003,Carol White,Monitor,2.0,15000.0,NaN,delivered
1004,Dave Brown,Keyboard,NaN,1200.0,2024-01-18,cancelled
1005,Eva Green,Laptop,1.0,85000.0,2024-01-19,delivered



Last 5 rows:


,customer_name,product,quantity,unit_price,order_date,status
order_id,,,,,,
1016,Paul Harris,Mouse,6.0,499.0,2024-01-30,pending
1017,Quinn Martin,Monitor,NaN,14500.0,2024-01-31,delivered
1018,Rachel King,Laptop,1.0,85000.0,NaN,delivered
1019,Sam Wright,Keyboard,3.0,1200.0,2024-02-01,cancelled
1020,Tina Baker,USB Hub,2.0,799.0,2024-02-02,pending


In [17]:
print("\n--- Structural Inspection ---")

# Shape
rows, cols = df.shape
print(f"\nRows: {rows}")
print(f"Columns: {cols}")

# Missing values
print("\nMissing values in each column:")
print(df.isnull().sum())

# Data types
print("\nData types before conversion:")
print(df.dtypes)

# Convert order_date to datetime
df['order_date'] = pd.to_datetime(df['order_date'])

print("\nData types after conversion:")
print(df.dtypes)


--- Structural Inspection ---

Rows: 20
Columns: 6

Missing values in each column:
customer_name    0
product          0
quantity         4
unit_price       0
order_date       4
status           0
dtype: int64

Data types before conversion:
customer_name     object
product           object
quantity         float64
unit_price       float64
order_date        object
status            object
dtype: object

Data types after conversion:
customer_name            object
product                  object
quantity                float64
unit_price              float64
order_date       datetime64[ns]
status                   object
dtype: object


#Task 2 — Structural Inspection

The dataset contains 20 rows and 6 columns.

The following columns have missing values:

quantity → 4 missing values

order_date → 4 missing values

The order_date column was initially stored as an object data type and was converted to datetime using:

pd.to_datetime(df['order_date'])

In [18]:
print("\n--- Statistical Summary ---")

summary = df.describe()
print(summary)

median_price = summary.loc['50%', 'unit_price']
mean_price = summary.loc['mean', 'unit_price']

print(f"\nMedian unit_price: ₹{median_price}")
print(f"Mean unit_price: ₹{mean_price}")

if mean_price > median_price:
    print("unit_price appears to be right-skewed (positively skewed).")
elif mean_price < median_price:
    print("unit_price appears to be left-skewed (negatively skewed).")
else:
    print("unit_price appears symmetric.")


--- Statistical Summary ---
        quantity    unit_price           order_date
count  16.000000     20.000000                   16
mean    2.437500  20224.600000  2024-01-24 09:00:00
min     1.000000    499.000000  2024-01-15 00:00:00
25%     1.000000    799.000000  2024-01-19 18:00:00
50%     2.000000   2500.000000  2024-01-24 12:00:00
75%     3.000000  14625.000000  2024-01-29 06:00:00
max     6.000000  85000.000000  2024-02-02 00:00:00
std     1.504161  33573.389607                  NaN

Median unit_price: ₹2500.0
Mean unit_price: ₹20224.6
unit_price appears to be right-skewed (positively skewed).


#Task 3 — Statistical Summary
The median unit_price is ₹2,500.

The unit_price distribution is right-skewed because:

The mean (₹20,224.60) is significantly greater than the median (₹2,500).

The maximum value (₹85,000) is much larger than the 75th percentile (₹14,625).

This indicates that high-value orders are increasing the average price.